frame/election-provider-support/src/onchain.rs

```diff
@@ -89,6 +90,12 @@ impl<T: Config> ElectionProvider<T::AccountId, T::BlockNumber> for OnChainSequen
 		let staked = assignment_ratio_to_staked_normalized(assignments, &stake_of)?;
 		let winners = to_without_backing(winners);
 
+		let weight = T::BlockWeights::get().max_block;
+		frame_system::Pallet::<T>::register_extra_weight_unchecked(
```

Comment:
This makes sense, but what where we doing for weight prior?

Reply:
we would return maximum block weight and return it in the `on_initialize` on the other side.